tripy/examples/resent50/resnet50.ipynb

```diff
@@ -0,0 +1,529 @@
+    "    \"\"\"\n",
+    "    This function loads weights from a state_dict into the custom TPResNet model,\n",
+    "    converting them into tp.Parameter before loading.\n",
+    "    \"\"\"\n",
+    "    model_dict = model.state_dict()  # Get the model's state dict\n",
+    "\n",
+    "    # Convert each weight to a tp.Parameter and update the model's state dict\n",
+    "    for k, v in pretrained_state_dict.items():\n",
+    "        if 'num_batches_tracked' in k:\n",
+    "            continue    # Skip num_batches_tracked since it is not needed for inference (tp.BatchNorm does not support)\n",
+    "        if 'classifier' in k:\n",
+    "            k = k.replace('1.', '') # Remove extra 1 (slight naming mismtach)\n",
+    "\n",
+    "        assert k in set(model_dict.keys())\n",
+    "        new_v = tp.Parameter(v.contiguous())\n",
+    "        model_dict[k] = new_v \n",
+    "\n",
+    "    model.load_state_dict(model_dict)\n",
+    "    return model\n",
+    "\n",
+    "pretrained_state_dict = resnet_pretrained.state_dict()\n",
+    "tripy_model = load_pretrained_weights_tp(tripy_model, pretrained_state_dict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we compile the model with static input shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_shape = [1, 3, 224, 224]\n",
+    "compiled_model = tp.compile(tripy_model, args=[tp.InputInfo(input_shape, dtype=tp.float32)])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run a dummy forward path\n",
+    "x = tp.ones(input_shape, dtype=tp.float32)\n",
+    "compiled_output = compiled_model(x)\n",
+    "compiled_output.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tripy Demo\n",
+    "\n",
+    "Now we will test our tripy resnet50 classification model on a few sample images for image classification task "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Prepare the image for Tripy\n",
+    "processed_image = processor(image, return_tensors=\"np\")['pixel_values']\n",
+    "tp_image = tp.Tensor(processed_image, dtype=tp.float32, device=tp.device(\"gpu\"))\n",
+    "tp_image.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run the image through the model\n",
+    "tp_logits = compiled_model(tp_image)\n",
+    "tp_logits.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get the predicted label\n",
+    "tp_predicted_label = torch.argmax(torch.from_dlpack(tp_logits), -1).tolist()[0]\n",
```

Comment:
Can we use `tp.argmax`?

Reply:
we can but @parthchadha suggested earlier to use torch for pre and post processing of input/outputs IIRC. Should I still change it to tp.argmax?

Reply:
you mean include tp.argmax in the model definition? so we don't return logits, we return `tp.argmax(logits)`?

Reply:
It doesn't have to be part of the modules, but it should be part of what we compile. So for example, you could have something like:
```py
def infer(x):
    logits = model(x)
    return tp.argmax(logits)

compiled_model = tp.compile(infer, ...)
```

Reply:
I see! that's great. Added it